model/detector_helpers.py

```diff
@@ -1,10 +1,21 @@
 from __future__ import absolute_import, division, print_function
 
+import itertools
 import math
 from builtins import object
+from operator import itemgetter
+
+import numpy as np
 
 from scitbx import matrix
 
+try:
+    import sklearn.cluster
+
+    have_sklearn = True
+except ImportError:
+    have_sklearn = False
```

Comment:
```suggestion
    import sklearn.cluster
except ImportError:
    sklearn = None
```
Alternatively, you can use `sklearn` as 'have' variable - `import sklearn.cluster` implicitly contains `import sklearn`

Reply:
Yes, that remains clear. I decided I didn't like using the name `cluster` for this (it got me in trouble after all), but `sklearn` is an unlikely variable name.